### ISSUE TITLE
Header / Sidebar Mobile Styles

### ISSUE THREAD

**Issue by MaximumCrash** [makerdao/community-portal#71]
Shea has provided us with a solution for the header and sidebar on mobile [here](https://www.figma.com/file/3Igwq7iPakOznIVpwwH1AA/SF_Maker-Comm-Dev?node-id=1170%3A344)

Images if you don't have access
<img width="818" alt="Screen Shot 2020-07-06 at 11 16 09 AM" src="https://user-images.githubusercontent.com/9157341/86610042-c75e9000-bf7a-11ea-8a62-942bdf1659dc.png">

~This execution will not require a complete rewrite of existing components. (Header and sidebar components are already segmented). However, the sidebar, only on mobile, will require some extra logic to render top sections as part of the transitioning logic.~ 

~There is a question currently posed for how we'll handle the experience of page changes (cause clicking a label should route you to the page) and the sliding transition of the sidebar on mobile. Awaiting an answer prior to starting.~

**Answered Below**

**Comment by MaximumCrash** [comment]
There was a conflict in the original spec provided by Shea where clicking a link meant traversing to a new page. Because of this she decided it'd be best if the sidenav is thought about in 2 ways.

1. The Main Menu
![Screen Shot 2020-07-06 at 11 21 49 AM](https://user-images.githubusercontent.com/9157341/86610184-faa11f00-bf7a-11ea-9201-8c0f34767f84.png)


This menu is visible at the top level. If someone opens the hamburger menu from the homepage they will see the image above. It is a carbon copy of the links provided in the header now. 

If the user is in a top level section like Work with us, Funding, or Learn, this will instead render a "sub-menu" (aka the current top section's Sidenav)

Clicking a link here will take you to the appropriate page and hide the hamburger menu. 

2. The Submenu
![Screen Shot 2020-07-06 at 11 22 09 AM](https://user-images.githubusercontent.com/9157341/86610198-ff65d300-bf7a-11ea-929a-a339cfe1d8d0.png)


This menu renders when our user is currently on a top level section (ie. Work with Us / Learn / ect.). If the user clicks a link here it will function **just like the sidenav** and take the user to the appropriate page. 

It is a carbon copy of the sidenav, except it offers a back tab at the top ("< Home") that can take the user back to the header related links. The label name is up for debate as "< Home" and "Home" signify a link to the homepage and can create confusion for what clicking it does at a glance. (Maybe "Main Menu"?)

**TLDR: Main Menu = Header | Sub Menu = Sidenav**

Acceptance Criteria:

- Create a hamburger mobile menu that meets the expectations of this spec. 
- When clicking on a link on **either** menu, take the user to the page and hide the hamburger menu.
- While in the submenu if the user taps  "< Home", the menu should update to render the Main Menu. 
- If the **submenu** is taller than the screen, ensure that when the menu is opened that the scroll position is to that of the currently active page. 
- When on the **main menu** and the user is currently on a page with a top level section, render it's link as expected (Bold, like Desktop).

**Comment by MaximumCrash** [comment]
Closed by #199
